lambdas/service/app.py

```diff
@@ -655,204 +661,237 @@ def repository_search(entity_type: str, item_id: str):
+                    param,
+                    schema.optional(str),
+                    description=format_description('''
+                        Use the `next` and `previous` properties of the
+                        `pagination` response element to navigate between pages.
+                    '''),
+                    deprecated=True)
+                for param in ['search_before', 'search_before_uid', 'search_after', 'search_after_uid']
+            ]
+        ],
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    Paginated list of {entity_type}s that meet the search
+                    criteria ("hits"). The structure of these hits is documented
+                    under the [corresponding endpoint for a specific entity]({id_spec_link}).
+
+                    The `pagination` section describes the total number of hits
+                    and total number of pages, as well as user-supplied search
+                    parameters for page size and sorting behavior. It also
+                    provides links for navigating forwards and backwards between
+                    pages of results.
+
+                    The `termFacets` section tabulates the occurrence of unique
+                    values within nested fields of the `hits` section. Not every
+                    nested field is tabulated, but the set of tabulated fields
+                    is consistent between entity types.
+                '''),
+                **schema.wrap_json_schema_content(page_spec)
+            }
+        }
+    }
+
+
+def repository_id_spec(entity_type: str):
+    search_spec_link = f'#operations-Repository-get_repository_{entity_type}s_'
+    return {
+        'summary': f'Detailed information on a particular {entity_type} entity.',
+        'tags': ['Repository'],
+        'parameters': [
+            params.path(f'{entity_type}_id', str, description=f'The UUID of the desired {entity_type}')
+        ],
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    This response describes a single {entity_type} entity. To
+                    search the repository for multiple entities, see the
+                    [corresponding search end endpoint]{search_spec_link}.
+
+                    The properties that are common to all entity types are
+                    listed in the schema below; however, additional properties
+                    may be present for certain entity types. With the exception
+                    of the {entity_type}'s unique identifier, all properties are
+                    arrays, even in cases where only one value is present.
+
+                    The inner structures of the objects within these arrays are
+                    not consistent between entity types, e.g. projects and
+                    files, and in a very small number of cases, not consistent
```

Comment:
What cases? We need to fix these inconsistencies.This is exactly the type of language that is sure to instil confusion among clients. If we can't fix them, we need to be more specific in documenting them.

Reply:
@hannes-ucsc Is this inconsistency worth documenting or should I remove my allusion to it?

Reply:
Samples are polymorphic. Any biomaterial that yields a cell suspension which yields a sequence file will be considered a *sample*. In the current HCA metadata schema, a sample can be either a specimen, an organoid or a cell line. Its content, when it occurs as an inner entity, depends on the concrete type of entity it represents. And that explains the variation you are observing. The discriminating property is `sampleEntityType`. Only on `/repository/samples` can we be sure that each`hits[*].samples` is a singleton and that its sole element represents an individual sample. For other repository endpoints, that document path—while still pointing at a a singleton list—represents an aggregate of potentially many samples that are all associated with the given outer entity represented by the hit. That's why `hits[*].samples[0].sampleEntityType` could be a list with more than one entry on those endpoints.

Reply:
Now that I put some effort into explaining this, please incorporate that explanation into the documentation so that our users benefit from it as well.